[Chart Enhancement] Fix Eggnog Erect ending notes + small adjustments

afreetoplaynoob opened the pull request:
The ending of Eggnog Erect was using the wrong BPM (80, instead of 70, half of the song's main speed) and the four ending notes were off-sync, so I fixed those things and made some other changes to the chart.

Fixes this in both Erect and Nightmare difficulties.

![Screenshot 2024-11-20 23 17 07](https://github.com/user-attachments/assets/c00bcc4b-0817-417c-ac55-38f4459cb293)

Hundrec commented on preload/data/songs/eggnog/eggnog-metadata-erect.json:
```diff
@@ -2,33 +2,72 @@
   "version": "2.2.4",
   "songName": "Eggnog Erect",
   "artist": "Saruky",
```
```suggestion
  "artist": "Saruky",
  "charter": "fabs + SpazKid",
```

Heads up, exporting charts removes the charter from the Metadata!

Hundrec commented on preload/data/songs/eggnog/eggnog-metadata-erect.json:
```diff
@@ -2,33 +2,72 @@
+    "difficulties": [
+      "erect",
+      "nightmare",
+      "hard"
```
```suggestion

```
Exporting charts from erect variations also creates an excess Hard difficulty with no notes.

Hundrec commented on preload/data/songs/eggnog/eggnog-chart-erect.json:
```diff
@@ -1,2086 +1,11166 @@
     }
   ],
   "notes": {
+    "hard": [    ],
```
```suggestion

```
Removing this excess Hard difficulty should make it good to go!

Hundrec commented:
No problem!
I've been curious: What is the full list of changes you made to the chart?
Ideally, we would be able to go in and manually edit the notes and events without re-exporting the song and making comparison difficult. (See line changes for #96 as an example)

afreetoplaynoob commented:
Excluding the aformentioned BPM change and ending notes:

## Section 11 - Parents (Christmas)
Added a 32nd down note to accurately chart the grace.

![Screenshot 2024-11-21 10 53 32](https://github.com/user-attachments/assets/a6d62a71-a2c8-4610-a150-b1faf6f3c897)

## Section 15 - Parents (Christmas)
Added a 32nd right note to accurately chart the grace.

![Screenshot 2024-11-21 10 54 27](https://github.com/user-attachments/assets/97949834-a675-4373-bfd2-cc13a2ceefeb)

## Section 47 - Parents (Christmas)
Added an up note for the small note Daddy Dearest sings here.

![Screenshot 2024-11-21 10 55 21](https://github.com/user-attachments/assets/8ad18c5f-0eff-4b63-9d4a-c6717f79e72f)

## Section 56 - Boyfriend (Christmas)
Removed the down note grace here and moved the right note a bit earlier.

![Screenshot 2024-11-21 11 01 02](https://github.com/user-attachments/assets/c4cb6731-b3d7-4ae3-93bd-040fceebce09)

## Section 30 - Parents (Christmas) [ERECT Exclusive]
Moved the three notes immediately after the event a bit lower to sync with the singing.

![Screenshot 2024-11-21 11 04 18](https://github.com/user-attachments/assets/708e2885-60a4-4472-8dcd-fce1f6a5f884)

## Note
The section 56 change isn't in the PR itself yet, as I only figured out the proper timing for the note when gathering screenshots for this comment.

Hundrec commented:
Yeah, the grid is super important!
I'll clarify that it looks like the new notes you've placed align in 32nds with the regular variation's 150bpm grid, affecting the accuracy of the placements. Please revise these placements on the correct grid and update me with the changes!

afreetoplaynoob commented:
Alright, here's the changes again, now in their _proper_ positions and sections, with the changed notes highlighted and already manually edited into the original file:

## Section 10 - Parents

![Screenshot 2024-11-21 20 26 23](https://github.com/user-attachments/assets/1e29c44f-b067-43c5-b6ac-9fb4e1756138)

## Section 14 - Parents

![Screenshot 2024-11-21 20 26 40](https://github.com/user-attachments/assets/2c9728fa-b64d-44df-9c26-17268649b9d5)

## Section 44 - Parents

![Screenshot 2024-11-21 20 26 57](https://github.com/user-attachments/assets/a4002c16-6bfc-4c93-83d4-c3ead2aeca9b)

## Section 52 - Boyfriend

![Screenshot 2024-11-21 20 27 06](https://github.com/user-attachments/assets/8af107d6-9a2b-4c6f-8b51-8fdac7be89f6)

## Section 71 - Both

Here's how it looks like in 70 BPM:
![Screenshot 2024-11-21 20 27 33](https://github.com/user-attachments/assets/20c340fa-256f-47a1-847e-d646afaa206a)
And how it looks like in 140 BPM, for easier viewing:
![Screenshot 2024-11-21 20 29 12](https://github.com/user-attachments/assets/fe4ebb8c-3a6f-4caa-abc8-93468197bf9b)

## Section 28 - Boyfriend [ERECT Only]

![Screenshot 2024-11-21 20 27 54](https://github.com/user-attachments/assets/9f8328e9-f3dc-41d6-ac03-7770895c53f5)

Hundrec commented:
Thanks so much! I'll look into these changes when I have the time and create a branch in my fork that manually edits the json files without re-exporting the chart.

That way we can use the GitHub diffs for comparison and save some disk space for the game!

Hundrec commented:
Oh whoops, didn't notice you had noted that you already made those manual edits.
Can you commit those changes to this PR so I can examine them?

Hundrec commented:
*Also* just noticed that you already made a commit for those manual edits.
I wonder why GitHub is displaying the diffs as if they're replacing entire chunks instead of just the parts you changed...

afreetoplaynoob commented:
Pretty sure it's from all the changes and addendums we've made to the file; this PR started with an exported chart file and then went back to the original file format, so it's seeing it as the diffs to the exported format.

Hundrec commented:
Got it, then I will manually copy over your files to my fork and see if the diffs are small there

Hundrec commented:
Just checked out your changes and they're great!
If you can squash this PR into one commit, it should look like [this branch](https://github.com/Hundrec/funkin.assets/commit/bc31b8983b222d9a7763d062c6fc1c1192fd337e) and be easily comparable.

afreetoplaynoob commented:
Looks like I miscalculated there, the diffs are still messed up...
At this point I might as well delete the repo and close this PR so I can make a new one with good diffs.
